src/state/registers.ts

```diff
@@ -583,6 +591,37 @@ export abstract class RegisterSet implements vscode.Disposable {
 
     this._named.set("", this.null);
     this._named.set("null", this.null);
+
+    // Watch systemClipboardRegister setting and update binding
+    if (extension !== undefined) {
+      extension.observePreference<keyof Registers>(
+        ".systemClipboardRegister",
+        (value) => {
+          if (!['"', null, ...availableClipboardRegisters].includes(value)) {
+            return;
+          }
+
+          // Reset old value to be a GeneralRegister
+          if (this._systemClipboardRegister === "dquote") {
+            this.dquote = new GeneralPurposeRegister('"', "copy");
+            this._named.set("dquote", this.dquote);
+          } else if (this._systemClipboardRegister != null) {
+            this._named.set(this._systemClipboardRegister, new GeneralPurposeRegister(this._systemClipboardRegister, "clippy"));
+          }
+
+          // Set new value to be a ClipboardRegister
+          if (value === "dquote") {
+            this.dquote = new ClipboardRegister();
```

Comment:
Perhaps you could rewrite the `dquote` field as a `get` property that reads `_named` to avoid keeping track of both `_named` and `dquote`?